salt/runner.py

```diff
@@ -256,7 +256,7 @@ def run(self):
                 # otherwise run it in the main process
                 if self.opts.get('eauth'):
                     ret = self.cmd_sync(low)
-                    if isinstance(ret, dict) and set(ret) == set(('data', 'outputter')):
+                    if isinstance(ret, dict) and set(ret) == {'data', 'outputter'}:
```

Comment:
Since `set(ret)` and `tuple(ret)` would both produce objects containing just the keys, this can be done just fine with tuples:

```python
if isinstance(ret, dict) and tuple(ret) == ('data', 'outputter'):

Reply:
Does order matters? Because set are unordered but tuples are.